### ISSUE TITLE
How to handle figures, tables et al.?

### ISSUE THREAD

**Issue by annettegessner** [OpenGreekAndLatin/First1KGreek#251]
In case of a work including some kind of figure or table, at this point we have to do the following:
Mark the textpart as a figure or [table](http://www.tei-c.org/release/doc/tei-p5-doc/fr/html/ref-table.html). Make sure you preserve the text inside the figure or table as accurately as possible and if necessary add a link to the exact PDF-page, where it can be found.

Example for figure with uncorrected text:
tlg4015.tlg001.opp-grc1
diagram on [page 28](https://archive.org/stream/commentariaina13pt12akaduoft#page/n49/mode/2up), 

Markup:

``` xml
<figure>
 <graphic url="https://archive.org/stream/commentariaina13pt12akaduoft#page/n49/mode/2up"/>
 <figDesc>οὐσία
συμβεβηκός
οὐκ ἐν
δποκειμένω
καὶ
καθ’ δπο
κειμενου
κα
θόλου
ἀσύστατον
μερ
ἐν ὕπο.
κειμένῳ
καὶ οὐ
καθ’ δπο-
κειμενου
ἷκόν</figDesc>
</figure>
```

BUT correct the text inside the figure (`<figDesc>`) by correcting the spelling and preserving the layout! 

At a later point, there will be [SVG](http://tei.it.ox.ac.uk/Talks/2007-10-31-MMWorkshop/P4toP5-1.xhtml) added to the XML.

**Comment by PonteIneptique** [comment]
Note : SVG can be really easily created with some tools online such as https://www.gliffy.com even if it is not their primarly goal (I think libre office Draw accepts also SVG as an output)

**Comment by sonofmun** [comment]
I don't think this will work for tables since, as far as I can tell (http://www.stoa.org/epidoc/gl/latest/app-elements.html) the `<table>` element is not allowed anywhere in EpiDoc. Perhaps I am missing something but this means we need to figure out a different way to deal with tables.

**Comment by lmuellner** [comment]
Hey Matt, maybe this has to change. There are going to have to be tables for some authors, like Ptolemy and others. Perhaps we talk about this at the meeting tomorrow? Best, Lenny

**Comment by sonofmun** [comment]
@simonastoyanova How would you handle this? E.g., isn't that big stone with all the tributes paid to Athens considered to be a table? If so, how is it encoded?

**Comment by LucieSty** [comment]
Hi ,

A quick note to all.  I will not be in the meetings on Tuesday.  I have a 2 day trip for UVA those two days.

Thanks,

— Lucie
Lucie Wall Stylianopoulos
Director of Academic Programs & Outreach
Art And PreModern Librarian
University of Virginia Library
Fiske Kimball Fine Arts Library
Box 400131
Charlottesville, VA 22901
Lucie@virginia.edu
434-924-6604

**Comment by planatheisa** [comment]
@sonofmun Suggestion for the tables: I replaced `<table>` with `<ab type="table">` (ab = anonymous block), `<tr>` (the rows) with `<list type="table">` and `<td>` (the cells) with `<item>`. So far, oxygen does not object. Haven't tried Hook yet. What do you think?

**Comment by sonofmun** [comment]
That is exactly the type of solution I think we will need. And if we decide we need to change it in the future, these places will be easy to find. Can you put a @type attribute on `<item>`? Something like @type="cell"? That would make those easier to find and easier to write CSS for in order to represent them correctly.

**Comment by planatheisa** [comment]
There is no @type but I could do @n. Also should I rather call the @type of `<list>` "row" instead of table?

**Comment by PonteIneptique** [comment]
Hey @planatheisa ,
from what you propose, it seems that
```xml
    <ab>
        <list rend="table">
            <item rend="column"><list rend="row"><item>Content</item></list></item>
        </list>
    </ab>
```
would be a good idea :)

**Comment by PonteIneptique** [comment]
And you actually would not need `<ab>` :)

**Comment by planatheisa** [comment]
And the cells just as `<items>` without attribute?

**Comment by planatheisa** [comment]
I'm not sure what the "column" would be for. It is not described in the xml and if you have the rows as list, it doesn't really add any info in my opinion.

**Comment by PonteIneptique** [comment]
Yup, because it makes sense these are cell. Putting rend="cell" would be quite expansive in bytes for big tables. Actually, column is not used in HTML so let's do this way .

If a cell span for more than one column or row, I would advise to use HTML "attributes" in item's style : 
```
<list rend="table">
  <list rend="row"><item>Content</item><item>Cell One</item></list>
  <list rend="row"><item style="colspan:2;">Content Spanning</item></list>
</list>
```

**Comment by PonteIneptique** [comment]
And we might use `rend="row-header"` and  `rend="row-footer"`if needed.

**Comment by PonteIneptique** [comment]
Happy we agree on Column ;)

**Comment by planatheisa** [comment]
OK, I'll try that.

**Comment by planatheisa** [comment]
So, you can't have a `<list>` within a `<list>`. Probably the reason for the `<item rend="column">`.

**Comment by planatheisa** [comment]
Can I return to `<ab>`. It seems unnecessary to have to add useless tags just to circumvent this.

**Comment by PonteIneptique** [comment]
So here is the proposal :D 
```xml
        <list rend="table">
            <item rend="header"><list rend="row"><item>Content</item></list></item>
            <item>
                <list rend="row"><item>Content</item><item>Cell One</item></list>
                <list rend="row"><item style="colspan:2;">Content Spanning</item></list>
            </item>
            <item rend="footer"><list rend="row"><item>Content</item></list></item>
        </list>
```

**Comment by PonteIneptique** [comment]
Notice that body has no rend, because it is the default.

This makes more sense than using `<ab />` because of the semanticless value of `<ab />`

**Comment by planatheisa** [comment]
I have numbered rows here, so the first cell of each row is a (greek) number. Should I add rend="header" or something similar to these?

**Comment by planatheisa** [comment]
Also, does each row (except for header and footer) have an individual `<item>` casing or do I put all in one `<item>`.

**Comment by planatheisa** [comment]
Example:

`<list rend="table">
    <item>
    <list rend="row">
       <item>α΄</item><item>μακρὸς</item><item>πλατὺς</item><item>ὑψηλὸς</item>
       <item>μέγας</item>
     </list>
     <list rend="row">
        <item>β΄</item><item>μακρὸς</item><item>πλατὺς</item><item>σύμμετρος</item>
      </list>
      <list rend="row">
         <item>γ΄</item><item>μακρὸς</item><item>πλατὺς</item>
         <item>ταπεινὸς</item>
       </list>
    </item>
</list>`

**Comment by PonteIneptique** [comment]
The main item acts like a body, which means there should be only one (so -> put all in one item)
I don't think it is necessary to put rend="header". But you could but these one in a `<label><num></num></label>` :)

```xml
<list rend="table">
    <item>
        <list rend="row"> <item><label><num>α΄</num></label></item><item>μακρὸς</item><item>πλατὺς</item><item>ὑψηλὸς</item> <item>μέγας</item> </list> 
        <list rend="row"> <item><label><num>β΄</num></label></item><item>μακρὸς</item><item>πλατὺς</item><item>σύμμετρος</item> </list>
        <list rend="row"> <item><label><num>γ΄</num></label></item><item>μακρὸς</item><item>πλατὺς</item> <item>ταπεινὸς</item> </list>
    </item> 
</list>
```

**Comment by PonteIneptique** [comment]
Note that there is one row which has one bonus item :) Not sure what this means without the original scan

**Comment by planatheisa** [comment]
[https://babel.hathitrust.org/cgi/pt?id=hvd.hxpp8p;view=1up;seq=514](url) 
Some have four elements in a row, some have three. There is no real explanation what the columns signifie. I mean, there probably is, but I can't read enough Latin or Greek. But thanks, I'll do that.

**Comment by lmuellner** [comment]
So this is the way to do a numbered table with a caption in EpiDoc? Seems insane, but hey...

**Comment by Pria20** [comment]
Is the plan still to modify the epidoc to allow for SVG's and/or <table>? In the meantime, should I go through and begin to flesh out all the <figure>'s and their descriptions?

**Comment by sonofmun** [comment]
@Pria20 Modifying epiDoc is the solution we are leaning towards, which would make both of these quite a bit easier.
If you are interested in going through and work on the figures, that would be great.

**Comment by annettegessner** [comment]
Hi guys,
Thank you for offering to take a look at the figures @Pria20! I am currently working on a file, that still has to be added to the First1kGreek repo (tlg0363.tlg001.1st1K-grc1.xml) and there quite a lot of figures due to it being on math. It seems there is no indication in the XML-file whatsoever, when there is a figure in the volume. Should I add some kind of (and which) annotation, where a figure has to be added, so it can be found?

**Comment by annettegessner** [comment]
Suggestion:
In the volume of pull request #1407, there are a lot of figures and tables in the text, which do not appear in the XML-file. I am going through it right now and am marking them up as follows:

For figures I give an annotation including the link to the PDF e.g.:
`<figure><graphic url="https://archive.org/stream/claudiiptolemae00ptolgoog#page/n92/mode/1up"/></figure>`
For tables I do the almost the same:
`<figure rend="table"><graphic url="https://archive.org/stream/claudiiptolemae00ptolgoog#page/n91/mode/1up"/></figure>`

Please let know, if this mark up is ok (for now)!

**Comment by sonofmun** [comment]
I think this looks really good!

**Comment by Pria20** [comment]
I'm marking them up much in the same way except I have been including `<figDesc>` as well

**Comment by annettegessner** [comment]
Great! And of course, yes, please add figDesc or anything necessary to the mark up!

**Comment by annettegessner** [comment]
Should I ask DDD to mark up figures as described above while working on the newly scanned texts? For example:
``<figure><graphic url="http://heml.mta.ca/lace/sidebysideview2/12871653"/></figure>``

**Comment by sonofmun** [comment]
Yes, please do. This will save us a lot of work later.

**Comment by annettegessner** [comment]
Ok!
